inference-engine/thirdparty/clDNN/runtime/debug_configuration.cpp

```diff
@@ -13,24 +15,83 @@ const char *debug_configuration::prefix = "GPU_Debug: ";
 // Default policy is that dump_configuration will override other configuration from IE.
 
 #ifdef GPU_DEBUG_CONFIG
-static void print_option(std::string option_name, std::string option_value) {
+
+template<typename T>
+void print_option(std::string option_name, T option_value) {
     GPU_DEBUG_COUT << "Config " << option_name << " = " << option_value << std::endl;
 }
 
-static void get_int_env(const std::string &var, int &val) {
-    if (const auto env_var = std::getenv(var.c_str())) {
-        val = std::stoi(env_var);
-        print_option(var, std::to_string(val));
+static std::string to_upper_case(const std::string& var) {
+    std::stringstream s;
+
+    for (size_t i = 0; i < var.size(); i++) {
+        if (std::isupper(var[i])) {
+            if (i != 0) {
+                s << "_";
+            }
+            s << var[i];
+        } else {
+            s << static_cast<char>(std::toupper(var[i]));
+        }
     }
+
+    return s.str();
+}
+
+static std::vector<std::string> get_possible_option_names(const std::string& var, std::vector<std::string> allowed_option_prefixes) {
+    std::vector<std::string> result;
+
+    for (auto& prefix : allowed_option_prefixes) {
+        result.push_back(prefix + var);
+        result.push_back(prefix + to_upper_case(var));
+    }
+
+    return result;
+}
+
+template <typename T>
+T convert_to(const std::string &str) {
+    std::istringstream ss(str);
+    T res;
+    ss >> res;
+    return res;
 }
 
-static void get_str_env(const std::string &var, std::string &val) {
-    if (const auto env_var = std::getenv(var.c_str())) {
-        val = env_var;
+template<typename T>
+void get_debug_env_var(const std::string &var, T &val, std::vector<std::string> allowed_option_prefixes) {
+    bool found = false;
+    for (auto o : get_possible_option_names(var, allowed_option_prefixes)) {
+        if (const auto env_var = std::getenv(o.c_str())) {
+            val = convert_to<T>(env_var);
+            found = true;
+        }
+    }
+
+    if (found) {
         print_option(var, val);
     }
 }
 
+template<typename T>
+void get_gpu_debug_env_var(const std::string &var, T &val) {
+    std::vector<std::string> allowed_option_prefixes = {
+        "OV_GPU_"
+    };
+
+    return get_debug_env_var(var, val, allowed_option_prefixes);
+}
+
+template<typename T>
+void get_common_debug_env_var(const std::string &var, T &val) {
+    // The list below should be prioritized from lowest to highest prefix priority
```

Comment:
Cannot understand intention of this comment. Could you elaborate more?

Reply:
Added better description